bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/cipher/SymmetricKeyCipher.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.auth.oauth2client.internal.cipher;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.auth.client.oauth2.StorageCipher;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a symmetric key encryption service for encrypting the OAuth tokens.
+ *
+ * @author Gary Tse - Initial Contribution
+ *
+ */
+@Component
+public class SymmetricKeyCipher implements StorageCipher {
+
+    public static final String CIPHER_ID = SymmetricKeyCipher.class.getName();
```

Comment:
Do you really think the FQCN is a proper identifier? This makes any future refactoring risky as it will easily break things in an incompatible way...

Reply:
good point.  Will just hard code it.

Reply:
... but hopefully not using the FQCN...